include/irr/asset/IImageFilter.h

```diff
@@ -16,20 +16,76 @@ namespace irr
 namespace asset
 {
 
-// runtime polymorphic
+//! Base class for general filters with runtime polymorphism.
+/*
+	Filters can execute various actions basing on input image
+	to get an output image.
+	Available filters are as following:
+
+	- Fill Filter
+	- Copy Filter
+	- Flatten Filter
+	- Convert Filter
+	- Swizzle && Convert Filter
+	- Utility Filter
+	- Blit Filter
+	- Generate Mip Maps Filter
+
+	Each of those performs certain specified by a filter actions
+	on an image's output attached texel buffer, where input image
+	is a reference for a valid process executation.
```

Comment:
actually there is complete freedom on filters inputs and outputs, this is why each defines (or at least typedefs) its own `state` 

There are input only, output only (such as fill) or input output filters.

Reply:
adjusted in following commit